project_euler/problem_234/sol1.py

```diff
@@ -17,40 +17,98 @@
 What is the sum of all semidivisible numbers not exceeding 999966663333 ?
 """
 
+import math
 
-def fib(a, b, n):
-
-    if n == 1:
-        return a
-    elif n == 2:
-        return b
-    elif n == 3:
-        return str(a) + str(b)
-
-    temp = 0
-    for x in range(2, n):
-        c = str(a) + str(b)
-        temp = b
-        b = c
-        a = temp
-    return c
-
-
-def solution(n):
-    """Returns the sum of all semidivisible numbers not exceeding n."""
-    semidivisible = []
-    for x in range(n):
-        l = [i for i in input().split()]  # noqa: E741
-        c2 = 1
-        while 1:
-            if len(fib(l[0], l[1], c2)) < int(l[2]):
-                c2 += 1
-            else:
+
+def sieve(n: int) -> list:
```

Comment:
```suggestion
def prime_sieve(n: int) -> list:
```
Please make the necessary changes elsewhere.